features/netsocket/InternetDatagram.h

```diff
@@ -0,0 +1,162 @@
+/** \addtogroup netsocket */
+/** @{*/
+/* InternetDatagram
+ * Copyright (c) 2015 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef InternetDatagram_H
+#define InternetDatagram_H
+
+#include "netsocket/InternetSocket.h"
+#include "netsocket/NetworkStack.h"
+#include "netsocket/NetworkInterface.h"
+#include "rtos/EventFlags.h"
+
+
+/** InternetDatagram socket implementation.
+ */
+class InternetDatagram : public InternetSocket {
+public:
+    /** Create an uninitialized socket.
+     *
+     *  @note Must call open to initialize the socket on a network stack.
+     */
+    InternetDatagram();
```

Comment:
Make this `= default`. Maybe should be protected to match `InternetSocket`, although wouldn't matter as a partially abstract class anyway (if you removed `ge_proto`).